karapace/kafka/consumer.py

```diff
@@ -86,7 +86,10 @@ def commit(  # type: ignore[override]
             if message is not None:
                 return super().commit(message=message, asynchronous=False)
 
-            return super().commit(offsets=offsets, asynchronous=False)
+            if offsets is not None:
+                return super().commit(offsets=offsets, asynchronous=False)
```

Comment:
Note: because this is a C library, there is a difference between providing `None` or not providing a parameter.

Reply:
oh this `is not None` being different from `!= None` and different from `bool(optional)` stole me so much time in my life 😭